benchmarks/microbenchmarks/_plan.py

```diff
@@ -62,3 +64,26 @@ def _get_kwargs(self, tags):
         for item, tag in zip(self.items, tags):
             kwargs[item.name] = item.impl(tag)
         return kwargs
+
+    def _get_item(self, name):
+        for item in self.items:
+            if item.name == name:
+                return item
+        return None
+
+    def _remove_comflict_items(self):
+        remove_list = []
+        item_name_list = [item.name for item in self.items]
+        #logical_atomic with float_type
+        if set([AtomicOps.name, DataType.name]).issubset(item_name_list):
+            atomic_item = self._get_item(AtomicOps.name)
+            dytpe_item = self._get_item(DataType.name)
+            for case, plan in self.plan.items():
+                atomic_op_tag = atomic_item.get_intersection(plan['tags'])
```

Comment:
Why not use a dictionary instead of list? If we use dictionary then we do not need to do the intersection.

Reply:
IIUC we can use the dictionary from `_get_kwargs` ?

Reply:
Thanks for the suggestion, the code looks much clearer in 42650e1

```
        if set([AtomicOps.name, DataType.name]).issubset(self.items.keys()):
            for name, case in self.plan.items():
                kwargs = self._get_kwargs(case['tags'])
                atomic_tag = self.items[AtomicOps.name].tag(kwargs[AtomicOps.name])
                dtype_tag = self.items[DataType.name].tag(kwargs[DataType.name])
                if not AtomicOps.is_supported_type(atomic_tag, dtype_tag):
                    remove_list.append(name)
```